### ISSUE TITLE
onDragEnd handler gets "unbound" when second Gmap and Marker created

### ISSUE THREAD

**Issue by gkrinc** [MicheleBertoli/react-gmaps#32]
I have a Gmap with a draggable Marker which triggers and action to update state in the onDragEnd event handler, which works great.

When the user double-clicks on the Gmap I open a second Gmap in a react-modal dialog that's larger. The second Marker is also draggable and works fine the first time. When the dialog is closed, the Marker on the first Gmap no longer triggers the onDragEng event, but the second Marker in the dialog Gmap works just fine.

I tried creating two distinct onDragEnd handlers to see if that would help and it didn't. I added names to the maps and markers to ensure everything was indeed distinct and that is true.

The onDragEnd appears to be "bound" according to the react tools before and after the dialog is opened/closed, but it acts differently after the dialog has been opened/closed.

Is there some way to rebind the onDragEnd handler?

<img width="321" alt="screen shot 2016-01-06 at 4 46 18 pm" src="https://cloud.githubusercontent.com/assets/2718382/12155813/37bfa37a-b495-11e5-9b95-d87cd2d61e4f.png">

```javascript
fullMapOpen: function(event){
        "use strict";

        this.setState({
            modalIsOpen: true
        });
    },

onMapMarkerDragEnd(event) {
        console.log('onDragEnd', event);
        var latLng = {};
        latLng[this.props.latKey] = event.latLng.lat();
        latLng[this.props.lngKey] = event.latLng.lng();
        this.props.onComponentChange(latLng);
    },


render: function(){
        "use strict";

        var availableWidth = (window.innerWidth * 0.86) + 'px';
        var availableHeight = (window.innerHeight * 0.86) + 'px';

        const modalStyles = {
            overlay: {
                backgroundColor: 'rgba(0,0,0,0.6)'
            },
            content : {
                top                   : '50%',
                left                  : '50%',
                right                 : 'auto',
                bottom                : 'auto',
                marginRight           : '-50%',
                transform             : 'translate(-50%, -50%)',
                padding               : '2px'
            }
        };

        return (
            <div id="map" onDoubleClick={this.fullMapOpen}>
                <Gmaps
                    name={'primaryAddressMap'}
                    width={'422px'}
                    height={'100px'}
                    lat={this.props.latitude}
                    lng={this.props.longitude}
                    zoom={17}
                    loadingMessage={'Loading...'}
                    params={{v: '3.exp', libraries: 'places'}}
                    onMapCreated={this.onMapCreated}
                    disableDoubleClickZoom={true}
                >
                    <Marker
                        name={'placeMarker'}
                        lat={this.props.latitude}
                        lng={this.props.longitude}
                        draggable={true}
                        onDragEnd={this.onMapMarkerDragEnd}
                    />
                </Gmaps>
                <Modal
                    isOpen={this.state.modalIsOpen}
                    onRequestClose={this.fullMapClose}
                    style={modalStyles} >
                        <div style={{position: 'absolute', zIndex: 100, top: 10, left: 0, right: 0, height: 0, textAlign: 'center'}}>
                            <input type="text" id="place-search" placeholder="Search" style={{ marginTop: 2, width: 300 }}/>
                        </div>
                        <Gmaps
                            name={'primaryAddressMapFull'}
                            width={availableWidth}
                            height={availableHeight}
                            lat={this.props.latitude}
                            lng={this.props.longitude}
                            zoom={17}
                            loadingMessage={'Loading...'}
                            params={{v: '3.exp'}}
                            onMapCreated={this.onMapCreated}
                            disableDoubleClickZoom={true}
                        >
                            <Marker
                                name={'placeMarkerFull'}
                                lat={this.props.latitude}
                                lng={this.props.longitude}
                                draggable={true}
                                onDragEnd={this.onMapMarkerDragEnd}
                            />
                        </Gmaps>
                </Modal>
            </div>
        );
    }
```

**Comment by MicheleBertoli** [comment]
Thanks @gkrinc, I'm looking into it.

**Comment by MicheleBertoli** [comment]
Thanks @gkrinc!
I managed to [reproduce](https://gist.github.com/MicheleBertoli/a973037d3a357634f8a4) the issue and I fixed it in [1.2.2](https://github.com/MicheleBertoli/react-gmaps/releases/tag/1.2.2).
It should now works as expected.
